src/main/java/io/vertx/httpproxy/impl/HttpProxyImpl.java

```diff
@@ -69,6 +63,14 @@ public void handle(HttpServerRequest outboundRequest) {
     });
   }
 
+  private Future<WebSocket> resolveOriginWS(HttpServerRequest outboundRequest) {
+    return selector.apply(outboundRequest).flatMap(server -> {
+      WebSocketConnectOptions webSocketConnectOptions = new WebSocketConnectOptions();
+      webSocketConnectOptions.setServer(server);
```

Comment:
we are missing stuff here like transmitting relevant HTTP headers from the client, some headers might contain important information like authentication

Reply:
Related code has been fixed.